airship_tempest_plugin/tests/api/shipyard/test_document_staging.py

```diff
@@ -0,0 +1,159 @@
+LOG = logging.getLogger(__name__)
+
+
+class DocumentStagingTest(base.BaseShipyardTest):
+    def _get_collection_id(self):
+        resp = self.shipyard_document_staging_client.get_configdocs_status()
+        self.assertTrue(len(resp[0]) > 0,
+                        'No configdocs available, nothing to test')
+        return resp[0]['collection_name']
+
+    @decorators.idempotent_id('db351ee5-a608-4492-b705-6e6c654a60e7')
+    def test_get_collection_list(self):
+        """Config docs status, Successful with response status 200"""
+        response = self.shipyard_document_staging_client.\
+            get_configdocs_status()
+        self.assertEqual(response.response['status'], '200')
+
+    @decorators.idempotent_id('743ab304-be35-4e45-ad47-e124dce3b9dc')
+    def test_get_collection(self):
+        """Get Config docs, Successful with response status 200
+           In case document does not found, return 404 Not Found
+        """
+        collection_id = self._get_collection_id()
+        try:
+            response = self.shipyard_document_staging_client.\
+                get_configdocs(collection_id)
+            self.assertEqual(response.response['status'], '200')
+        except exceptions.NotFound:
+            print("The Shipyard buffer does not contain this collection")
+            pass
+
+    @decorators.idempotent_id('bd138747-582e-4c2e-9f26-09c3fccad96f')
+    def test_get_collection_version_buffer(self):
+        """Get Config docs of buffer version,
+           Successful with response status 200
+           In case document does not found, return 404 Not Found
+        """
+        collection_id = self._get_collection_id()
+        try:
+            response = self.shipyard_document_staging_client.\
+                get_configdocs_version(collection_id, "buffer")
+            self.assertEqual(response.response['status'], '200')
+        except exceptions.NotFound:
+            print("The Shipyard buffer does not contain this collection")
+            pass
+
+    @decorators.idempotent_id('d42c691a-07a2-41cc-91a6-9d02be8f2649')
+    def test_get_collection_version_committed(self):
+        """Get Config docs of committed version,
+           Successful with response status 200
+        """
+        collection_id = self._get_collection_id()
+        response = self.shipyard_document_staging_client.\
+            get_configdocs_version(collection_id, "committed")
+        self.assertEqual(response.response['status'], '200')
+
+    @decorators.idempotent_id('2940492a-47b0-4218-859a-b9cb556f480b')
+    def test_get_collection_version_successful_site_action(self):
+        """Get Config docs of successful_site_action version,
+        Successful with response status 200
+        """
+        collection_id = self._get_collection_id()
+        response = self.shipyard_document_staging_client.\
+            get_configdocs_version(collection_id, "successful_site_action")
+        self.assertEqual(response.response['status'], '200')
```

Comment:
same comment as above; "just curious why you selectively use the try/except logic when collection_id is being passed as an argument"

Reply:
In this scenario, "404 NotFound exception" is not returned; therefore "NotFound exception is not used. 
As per Shipyard API document, wherever "404 NotFound Exception" may be returned because of unavailability of the intended result(like collection, buffer etc), try/exception is used for those scenarios only.